### ISSUE TITLE
Provide option to use actual OpenAI Backends.

### ISSUE THREAD

**Issue by prasann** [Azure-Samples/apim-genai-gateway-toolkit#10]
Currently, `USE_SIMULATOR` flag will control the deployment of the simulator. However, when we set the value to `false` it breaks the deployment due to simulator_key. 

There are few options,
- 1. Along with URLs as env, get their corresponding keys as well from `.env`
- 2. Make `simulator_key` as optional, provide guidance to use `api_key` as a header

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

**Comment by rajesh-ms** [comment]
Any update on this issue ? Any workaround ?

**Comment by lucashuet93** [comment]
Closed via PR #63
